src/components/toasts/TestnetToast.js

```diff
@@ -1,21 +1,24 @@
 import React, { useEffect, useState } from 'react';
-import networkTypes from '../../helpers/networkTypes';
 import { Icon } from '../icons';
 import { Nbsp, Text } from '../text';
 import Toast from './Toast';
 import { isHardHat } from '@/handlers/web3';
 import { useInternetStatus } from '@/hooks';
-import { getNetworkObj } from '@/networks';
+import { getNetworkObject } from '@/networks';
+import { ChainId } from '@/networks/types';
+import { useConnectedToHardhatStore } from '@/state/connectedToHardhat';
 
-const TestnetToast = ({ network, web3Provider }) => {
+const TestnetToast = ({ chainId }) => {
+  const { connectedToHardhat } = useConnectedToHardhatStore();
   const isConnected = useInternetStatus();
-  const providerUrl = web3Provider?.connection?.url;
-  const { name, colors: networkColors } = getNetworkObj(network);
-  const [visible, setVisible] = useState(!network === networkTypes.mainnet);
+  const { name, colors: networkColors } = getNetworkObject({ chainId });
+  const [visible, setVisible] = useState(true);
   const [networkName, setNetworkName] = useState(name);
 
   useEffect(() => {
-    if (network === networkTypes.mainnet) {
+    if ((connectedToHardhat, chainId === ChainId.mainnet)) {
```

Comment:
are you doing this just to use `connectedToHardhat` so we can have the `useEffect` depend on `connectedToHardhat` without dealing with warnings / autodeleting the dependency?

Reply:
yes but it's wrong like this, not sure what happened, good catch

connectedToHardhat && chainId === ChainId.mainnet

changed to that

Reply:
ah so we want to replace the check using `isHardHat` with `connectedToHardhat` because now this won't work for mainnet that is not connected to Hardhat (it'll go to the `else` which is the path for testnets)